examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py

```diff
@@ -35,311 +39,399 @@ def __init__(self, kv_helper):
-        input_value['result']['updater'] = {}
-        input_json_str = json.dumps(input_value)
-
-        if(self.kv_helper.get("wo-receipts", wo_id) is None):
-            value = self.kv_helper.get("wo-requests", wo_id)
-            if value:
-                self.kv_helper.set("wo-receipts", wo_id, input_json_str)
-                response = utility.create_error_response(
-                    WorkOrderStatus.SUCCESS, jrpc_id,
-                    "Receipt created successfully")
-            else:
-                response = utility.create_error_response(
-                    WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE, jrpc_id,
-                    "Work order does not exists. Hence invalid parameter")
+        wo_request = self.kv_helper.get("wo-requests", wo_id)
+        if wo_request is None:
+            raise JSONRPCDispatchException(
+                JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                "Work order id does not exist, "
+                "hence invalid request"
+            )
         else:
-            response = utility.create_error_response(
-                WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE, jrpc_id,
-                "Work order receipt already exist in the database. Hence invalid parameter")
+            wo_receipt = self.kv_helper.get("wo-receipts", wo_id)
+            if wo_receipt is None:
+                status, err_msg = self.__validate_work_order_receipt_create_req(
+                    input_value, wo_request
+                )
+                if status == True:
+                    self.kv_helper.set("wo-receipts", wo_id, input_json_str)
+                    raise JSONRPCDispatchException(
+                        JRPCErrorCodes.SUCCESS,
+                        "Receipt created successfully"
+                    )
+                else:
+                    raise JSONRPCDispatchException(
+                        JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                        err_msg
+                    )
+            else:
+                raise JSONRPCDispatchException(
+                    JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                    "Work order receipt is already exists. Hence invalid parameter"
+                )
+
+# ------------------------------------------------------------------------------------------------
+
+    def __validate_work_order_receipt_create_req(self, wo_receipt_req, wo_request):
+        """
+        Function to validate the work order receipt create request parameters
+        Parameters:
+            - wo_receipt_req is work order receipt request as dictionary
+            - wo_request is string containing jrpc work order request
+        Returns - tuple containing validation status(Boolean) and error message(string)
+        """
+        # Valid parameters list
+        valid_params = ["workOrderId", "workerServiceId", "workerId", "requesterId",
```

Comment:
We should consider putting all such strings to a constants file or declare at the top of a py file otherwise. Could be taken up as a separate PR for the complete codebase.